src/index.ts

```diff
@@ -53,193 +55,32 @@ router.use(function(req, res, next) {
-  currencyA: string,
-  currencyB: string,
-  date: string,
-  log: Function
-): Promise<string> {
-  let rate = ''
-  try {
-    const exchangeRate: nano.DocumentGetResponse & {
-      [pair: string]: any
-    } = await dbRates.get(date)
-    if (exchangeRate[`${currencyA}_${currencyB}`] != null) {
-      rate = exchangeRate[`${currencyA}_${currencyB}`]
-    } else if (exchangeRate[`${currencyB}_${currencyA}`] != null) {
-      rate = bns.div('1', exchangeRate[`${currencyB}_${currencyA}`], 8, 10)
-    }
-  } catch (e) {
-    if (e.error !== 'not_found') {
-      log(`DB read error ${JSON.stringify(e)}`)
-      throw e
-    }
-  }
-  return rate
-}
-
-const zeroRateCurrencyCodes = {
-  UFO: true,
-  FORK: true
-}
-
-const fallbackConstantRatePairs = {
-  SAI_USD: 1,
-  DAI_USD: 1
-}
-
 /*
  * Query params:
  * currency_pair: String with the two currencies separated by an underscore. Ex: "ETH_USD"
  */
 router.get('/exchangeRate', async function(req, res) {
-  let dateStr: string
-  const currencyPair = req.query.currency_pair
-  if (typeof req.query.date === 'string') {
-    dateStr = req.query.date
-  } else {
-    dateStr = new Date().toISOString()
-  }
-  if (typeof currencyPair !== 'string' || typeof dateStr !== 'string') {
-    return res.status(400).json({
-      error:
-        'Missing or invalid query param(s): currency_pair and date should both be strings'
-    })
-  }
-  const currencyTokens = currencyPair.split('_')
-  if (currencyTokens.length !== 2) {
-    return res.status(400).json({
-      error:
-        'currency_pair query param malformed.  should be [curA]_[curB], ex: "ETH_USD"'
-    })
-  }
-  const currencyA = currencyTokens[0]
-  const currencyB = currencyTokens[1]
-  const dateNorm = normalizeDate(currencyA, currencyB, dateStr)
-  if (dateNorm == null) {
-    return res.status(400).json({
-      error:
-        'date query param malformed.  should be conventional date string, ex:"2019-11-21T15:28:21.123Z"'
-    })
-  }
-  if (Date.parse(dateNorm) > Date.now()) {
-    return res
-      .status(400)
-      .json({ error: 'Future date received. Must send past date.' })
-  }
-  const log = (...args): void => {
-    const d = new Date().toISOString()
-    const p = currencyPair
-    console.log(`${d} ${p} ${JSON.stringify(args)}`)
-  }
-  let rate = ''
-  if (
-    zeroRateCurrencyCodes[currencyA] === true ||
-    zeroRateCurrencyCodes[currencyB] === true
-  ) {
-    rate = '0'
-  }
-  try {
-    let existingDocument
-    if (rate === '') {
-      rate = await getFromDb(currencyA, currencyB, dateNorm, log)
-    }
-    if (rate === '') {
-      existingDocument = await dbRates.get(dateNorm).catch(e => {
-        if (e.error !== 'not_found')
-          log(`DB existing doc read error ${JSON.stringify(e)}`)
-      })
-      if (existingDocument == null) {
-        existingDocument = {
-          _id: dateNorm,
-          [currencyPair]: rate
-        }
-      }
+  res.json(await getExchangeRate(req.query, dbRates))
```

Comment:
you need to "try-catch" here and return correct status code if error similar to what you are doing in "exchangeRates"